### ISSUE TITLE
[Concept] Thor

### ISSUE THREAD

**Issue by kilrau** [ExchangeUnion/xud#425]
We need a "thor"-like ("Flash Boys") implementation at some point where we
* measure pings to nodes
* delay orders for them to arrive simultaneously at target xud's

Reason: front running

Advanced, so clearly after `1.0.0`...

**Comment by hatmer** [comment]
I am working on this issue, please assign to me.

**Comment by kilrau** [comment]
You got it! Definitely not an easy feat! Feel free to ask for a "concept review" and any questions on the way!

**Comment by kilrau** [comment]
@hatmer already made progress on this? For any questions, feel free to ask here or in https://discord.gg/YgDhMSn

**Comment by hatmer** [comment]
Yes I have a working version [here](https://github.com/hatmer/xud/tree/feat/THOR), but it is vulnerable to malicious xud operators who modify their OS to be artificially slow. I have a new design that avoids this and plan to implement it today.
